pkg/master/master.go

```diff
@@ -781,11 +782,14 @@ func (m *Master) api_v1() *apiserver.APIGroupVersion {
 func (m *Master) expapi(c *Config) *apiserver.APIGroupVersion {
 	controllerStorage := expcontrolleretcd.NewStorage(c.DatabaseStorage)
 	autoscalerStorage := horizontalpodautoscaleretcd.NewREST(c.DatabaseStorage)
+	nodeMetricsStorage := nodemetrics.NewREST(c.DatabaseStorage)
 
 	storage := map[string]rest.Storage{
 		strings.ToLower("replicationControllers"):       controllerStorage.ReplicationController,
 		strings.ToLower("replicationControllers/scale"): controllerStorage.Scale,
 		strings.ToLower("horizontalpodautoscalers"):     autoscalerStorage,
+		// /api/experimental/nodeMetrics/{myNode}
```

Comment:
The naming rules for storage (what's exposed here) and the prefix key in
etcd (what's in /etcd.go) are two different cases.  Make this key look like
the other keys.

On Fri, Aug 21, 2015 at 4:52 PM, Daniel Martí <notifications@github.com>
wrote:

> In pkg/master/master.go
> <https://github.com/kubernetes/kubernetes/pull/12952#discussion_r37677002>
> :
>
> >
> >  	storage := map[string]rest.Storage{
> >  		strings.ToLower("replicationControllers"):       controllerStorage.ReplicationController,
> >  		strings.ToLower("replicationControllers/scale"): controllerStorage.Scale,
> >  		strings.ToLower("horizontalpodautoscalers"):     autoscalerStorage,
> > +		// /api/experimental/nodeMetrics/{myNode}
>
> The comment was just to remind myself, to be removed once the PR is ready.
>
> horizontalpodautoscalers was merged not long ago, see that PR for
> comments on the naming. Specifically:
>
> #12344 (comment)
> <https://github.com/kubernetes/kubernetes/pull/12344#discussion_r36701059>
>
> I just followed that pattern.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12952/files#r37677002>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift